tests/cases/fourslash/refactorInferFunctionReturnType16.ts

```diff
@@ -0,0 +1,17 @@
+/// <reference path='fourslash.ts' />
+
+////function /*a*/f/*c*/()/*d*//*b*/ /*e*/{
+////    /*g*/return { x: 1, y: 1 };/*h*/
+////}/*f*/
+
+goTo.select("a", "b");

```

Comment:
Can we have a few tests (you can reuse these files) for positions rather than ranges?

Reply:
Do you mean 1 selection range per file?

Reply:
You can use zero width spans for checking if the refactor is offered when the cursor is at a position but no span is selected,

```ts
////function f1(/*a*//*b*/x, y) {
////    return x + y;
////}
```

There are some verifiers for trigger reason if you want different behavior when `ctrl` + `shift` + `r` is invoked, but I don't think there is a case for that here (maybe on the return statement?)
`verify.not.refactorAvailableForTriggerReason("implicit", Infer function return type);`
`verify.refactorAvailableForTriggerReason("invoked", Infer function return type);`

Reply:
@jessetrinity Thanks for the detailed clarification. ;), I didn't know about it.